src/Lean/Elab/PreDefinition/WF/Preprocess.lean

```diff
@@ -110,14 +110,68 @@ builtin_dsimproc paramMatcher (_) := fun e => do
   let matcherApp' := { matcherApp with discrs := discrs', alts := alts' }
   return .continue <| matcherApp'.toExpr
 
-/-- `let x := (wfParam e); body[x] ==> let x := e; body[wfParam y] -/
+private def anyLetValueIsWfParam (e : Expr) : Bool :=
```

Comment:
@nomeata I am assuming you are happy with these changes. I am asking because you own this file.